data/store/mongo/mongo_test.go

```diff
@@ -279,12 +301,48 @@ var _ = Describe("Mongo", func() {
 					dataSet = NewDataSet(userID, deviceID)
 				})
 
+				Context("DateUnMarshal", func() {
+					var legacyUpload *dataTypesUploadTest.LegacyUpload
+					var result *upload.Upload
+					var createdTime time.Time
+					var modifiedTime time.Time
+					var deletedTime time.Time
+					var recordTime time.Time
+
+					BeforeEach(func() {
+						legacyUpload = NewLegacyDataSet(userID, deviceID)
+						createdTime, _ = time.Parse(time.RFC3339, "2016-09-01T11:00:00Z")
+						modifiedTime, _ = time.Parse(time.RFC3339, "2016-10-01T11:00:00Z")
```

Comment:
The source string doesn't have nanos, which is what we would store in the database.

Instead of parsing a static string time, you can just generate a random time with the test utils, e.g.: `test.PastFarTime()`.